src/Models/IServiceList.ts

```diff
@@ -0,0 +1,61 @@
+import { Model } from './Model';
+import { IParameter, Parameter } from './IParameters';
+
+export interface IService {
+    name: string;
+    action?: string;
+    version?: number;
+    parameters?: IParameter[];
+}
+
+export class Service extends Model implements IService {
+    constructor(data: IService) {
+        super(data);
+    }
+
+    set action(value: string) {
+        this.set('action', value);
+    }
```

Comment:
All classes that extend the model class can have their properties values be set in 2 ways either directly after you create an instance, or if you pass an object in the constructor that has a parameter with the same name as that property it will take the value of that property, this happends in the model class setOwnProperties in order for the class to know that the property can be set through the constructor we have to define it with a setter